jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java

```diff
@@ -197,13 +197,13 @@ public static Booster train(
           metrics[i][iter] = metricsOut[i];
         }
 
-        boolean decreasing = true;
+        boolean decreasing = false;
         float[] criterion = metrics[metrics.length - 1];
         for (int shift = 0; shift < Math.min(iter, earlyStoppingRound) - 1; shift++) {
-          decreasing &= criterion[iter - shift] <= criterion[iter - shift - 1];
+          decreasing |= criterion[iter - shift] <= criterion[iter - shift - 1];
```

Comment:
I'm not sure this is correct. The training must stop after `earlyStoppingRound` consecutive increases in the criterion. Your PR changes the semantics to *at least a single decrease*.

Reply:
if there is at least one decrease, it means it is not increasing throughout the consecutive period. Pay attention to the fact that it checks "!decreasing" to stop early. So if it is decreasing, then it is fine, meaning if it found at least one decreasing round, then continue. If all rounds are strictly increasing, then it is overfitting and you should stop. That's the logic.